src/main/java/spireMapOverhaul/zones/gremlinTown/GremlinTown.java

```diff
@@ -0,0 +1,652 @@
+            case "GremlinHealer":
+                return new GremlinHealer(xPos, yPos);
+            case "ChubbyGremlin":
+                return new ChubbyGremlin(xPos, yPos);
+            case "ArmoredGremlin":
+                return new ArmoredGremlin(xPos, yPos);
+            default:
+                return new GremlinWarrior(xPos, yPos);
+        }
+    }
+
+    @Override
+    public List<ZoneEncounter> getEliteEncounters() {
+        // Override that one method so the gremlin leader encounter isn't double registered
+        return Arrays.asList(
+                GREMLIN_LEADER_ENCOUNTER,
+                new ZoneEncounter(GREMLIN_ELDER, 2, () -> new MonsterGroup(new GremlinElder(-80f, -50f))),
+                new ZoneEncounter(GREMLIN_HORDE, 2, GremlinTown::getHordeStartGroup)
+        );
+    }
+
+    private static MonsterGroup getHordeStartGroup() {
+        ArrayList<String> gremlinPool = new ArrayList<>();
+        gremlinPool.add("GremlinWarrior");
+        gremlinPool.add("GremlinThief");
+        gremlinPool.add("GremlinFat");
+        gremlinPool.add("GremlinTsundere");
+
+        AbstractMonster[] retVal = new AbstractMonster[4];
+        int index = AbstractDungeon.miscRng.random(gremlinPool.size() - 1);
+        String key = gremlinPool.get(index);
+        gremlinPool.remove(index);
+        retVal[0] = getHordeGremlin(key, -320.0F, 25.0F);
+        index = AbstractDungeon.miscRng.random(gremlinPool.size() - 1);
+        key = gremlinPool.get(index);
+        gremlinPool.remove(index);
+        retVal[1] = getHordeGremlin(key, -160.0F, -12.0F);
+        index = AbstractDungeon.miscRng.random(gremlinPool.size() - 1);
+        key = gremlinPool.get(index);
+        gremlinPool.remove(index);
+        retVal[2] = getHordeGremlin(key, 25.0F, -35.0F);
+        index = AbstractDungeon.miscRng.random(gremlinPool.size() - 1);
+        key = gremlinPool.get(index);
+        gremlinPool.remove(index);
+        retVal[3] = getHordeGremlin(key, 205.0F, 40.0F);
+
+        return new MonsterGroup(retVal);
+    }
+
+    public static AbstractMonster getHordeGremlin(String key, float xPos, float yPos) {
+        switch (key) {
+            case "GremlinWarrior":
+                return new GremlinWarrior(xPos, yPos);
+            case "GremlinThief":
+                return new GremlinThief(xPos, yPos);
+            case "GremlinFat":
+                return new GremlinFat(xPos, yPos);
+            case "GremlinTsundere":
+                return new GremlinTsundere(xPos, yPos);
+            default:
+                return new GremlinWizard(xPos, yPos);
+        }
+    }
+
+    @Override
+    public void registerEncounters() {
+        List<ZoneEncounter> normalEncounters = this.getNormalEncounters();
+        List<ZoneEncounter> encounters = new ArrayList<>();
+        if (normalEncounters != null)
+            encounters.addAll(normalEncounters);
+
+        //  Will need to manually add all Elite encounters that aren't leader
```

Comment:
I believe I see why -- the direct reuse of the Gremlin Leader encounter (with the same ID as the base game) would probably make for some kind of conflict otherwise.

That said, you could probably avoid any conflict by registering a Gremlin Leader fight with a different ID, so it's technically speaking a different encounter, even though the contents are the same.

Reply:
Eh, this works, and doesn't put a duplicate fight in the dev console's list of encounters.